# In function ‘snprintf’, inlined from ‘read_patch’ at patch.c:344:4:
# warning: call to __builtin___snprintf_chk will always overflow destination buffer
#
# This is caused by this statements:
#
# | char new_file_name[PATH_MAX];
# | ...
# | snprintf(new_file_name, sizeof(char *) * PATH_MAX, ...
#
# where: sizeof(new_file_name) < sizeof(char *) * PATH_MAX
#

--- phasex.orig/src/patch.c	2014-08-30 05:35:44.775912325 +0200
+++ phasex/src/patch.c	2014-08-30 05:35:07.534910319 +0200
@@ -312,7 +312,7 @@
 int
 read_patch(char *filename, PATCH *patch)
 {
-	char            new_file_name[PATH_MAX];
+	char            new_file_name[PATH_MAX+1];
 	char            lash_patch_filename[16];
 	PARAM           *param;
 	DIR_LIST        *pdir       = patch_dir_list;
@@ -342,7 +342,7 @@
 	if (index(filename, '/') == NULL) {
 		while ((pdir != NULL) && (!dir_found)) {
 			snprintf(new_file_name,
-			         sizeof(char *) * PATH_MAX,
+			         PATH_MAX,
 			         "%s/%s.phx",
 			         pdir->name,
 			         filename);
